xarray/core/dataset.py

```diff
@@ -6651,6 +6681,20 @@ def pad(
                     end_values=end_values,
                     reflect_type=reflect_type,
                 )
+            elif name in var_pad_width.keys() and not isinstance(
+                var_pad_width[name], int
+            ):  # dimension coordinates
+                w0, w1 = pad_width[name]
+                fill_value_ind = dtypes.get_fill_value(var.dtype)
+                if isinstance(w0, int):
+                    w0 = IndexVariable(name, [fill_value_ind] * w0)
```

Comment:
```suggestion
                    w0_ = IndexVariable(name, [fill_value_ind] * w0)
```
That is a redefinition, so we need a new name (or a cast).